rts/motoko-rts/src/idl.rs

```diff
@@ -543,8 +575,356 @@ unsafe fn is_opt_reserved(typtbl: *mut *mut u8, end: *mut u8, t: i32) -> bool {
+                match read_byte(&mut tb1) {
+                    1 => a11 = true,
+                    2 => a12 = true,
+                    3 => a13 = true,
+                    _ => {}
+                }
+            }
+            let mut a21 = false;
+            let mut a22 = false;
+            let mut a23 = false;
+            for _ in 0..leb128_decode(&mut tb2) {
+                match read_byte(&mut tb2) {
+                    1 => a21 = true,
+                    2 => a22 = true,
+                    3 => a23 = true,
+                    _ => {}
+                }
+            }
+            a11 == a21 && a12 == a22 && a13 == a23
+        }
+        (IDL_EXT_tuple, IDL_EXT_tuple) => {
+            let n1 = leb128_decode(&mut tb1);
+            let n2 = leb128_decode(&mut tb2);
+            if n1 != n2 {
+                return false;
+            }
+            for _ in 0..n1 {
+                let tag1 = leb128_decode(&mut tb1);
+                let t11 = sleb128_decode(&mut tb1);
+                let tag2 = leb128_decode(&mut tb2);
+                let t21 = sleb128_decode(&mut tb2);
+                if tag1 != tag2 {
+                    return false;
+                }
+                if !memory_compatible(rel, variance, typtbl1, typtbl2, end1, end2, t11, t21, false)
+                {
+                    return false;
+                }
+            }
+            true
+        }
+        (IDL_CON_record, IDL_CON_record) => {
+            let mut n1 = leb128_decode(&mut tb1);
+            let n2 = leb128_decode(&mut tb2);
+            let mut tag1 = 0;
+            let mut t11 = 0;
+            let mut advance = true;
+            for _ in 0..n2 {
+                let tag2 = leb128_decode(&mut tb2);
+                let t21 = sleb128_decode(&mut tb2);
+                if n1 == 0 {
+                    // Optional additional fields are only supported in the main actor type.
+                    if variance == TypeVariance::Invariance
+                        || !main_actor
+                        || !is_opt_reserved(typtbl2, end2, t21)
+                    {
+                        return false;
+                    }
+                    continue;
+                };
+                if advance {
+                    loop {
+                        tag1 = leb128_decode(&mut tb1);
+                        t11 = sleb128_decode(&mut tb1);
+                        n1 -= 1;
+                        // Do not skip fields during invariance check.
+                        if variance == TypeVariance::Invariance || !(tag1 < tag2 && n1 > 0) {
+                            break;
+                        }
+                    }
+                };
+                if tag1 > tag2 {
+                    // Optional additional fields are only supported in the main actor type.
+                    if variance == TypeVariance::Invariance
+                        || !main_actor
+                        || !is_opt_reserved(typtbl2, end2, t21)
```

Comment:
```suggestion
```
I don't think we need this check - its a candid hangover. It's enough to allow additional fields in the main actor.

Reply:
I agree. Thanks for the suggestion. I changed the code accordingly (and updated the comment).